### ISSUE TITLE
MacOS 10.5 20201011.1 pip package manager

### ISSUE THREAD

**Issue by jngrad** [actions/virtual-environments#1819]
**Description**  
Installing python packages with `pip3` can fail in two ways:
* the `vtk` package cannot be found for python 3.8
* packages are installed for python 3.9 (happens randomly)

Related to #1818. Originally reported in espressomd/espresso#3950.

**Area for Triage**:  Python

**Question, Bug, or Feature?**:  Bug

**Virtual environments affected**  
- [x] macOS 10.15
- [ ] Ubuntu 16.04 LTS
- [ ] Ubuntu 18.04 LTS
- [ ] Ubuntu 20.04 LTS
- [ ] Windows Server 2016 R2
- [ ] Windows Server 2019

**Expected behavior**  

`pip3` should find and install python 3.8 packages.

From https://github.com/espressomd/espresso/runs/1243427136 that uses 20201003.1:
```
Collecting numpy
  Downloading numpy-1.19.2-cp38-cp38-macosx_10_9_x86_64.whl (15.3 MB)
Collecting cython
  Downloading Cython-0.29.21-cp38-cp38-macosx_10_9_x86_64.whl (1.9 MB)
Collecting vtk
  Downloading vtk-9.0.1-cp38-cp38-macosx_10_9_x86_64.whl (80.7 MB)
Collecting h5py
  Downloading h5py-2.10.0-cp38-cp38-macosx_10_9_x86_64.whl (3.0 MB)
Collecting scipy
  Downloading scipy-1.5.2-cp38-cp38-macosx_10_9_x86_64.whl (28.9 MB)
Collecting six
  Downloading six-1.15.0-py2.py3-none-any.whl (10 kB)
Installing collected packages: numpy, cython, vtk, six, h5py, scipy
Successfully installed cython-0.29.21 h5py-2.10.0 numpy-1.19.2 scipy-1.5.2 six-1.15.0 vtk-9.0.1
```

**Actual behavior**  

Fails to download the `vtk-9.0.1-cp38-cp38-macosx_10_9_x86_64.whl` wheel (https://pypi.org/project/vtk/#files).

From https://github.com/espressomd/espresso/runs/1254313147
```
Collecting numpy
  Downloading numpy-1.19.2.zip (7.3 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Collecting cython
  Using cached Cython-0.29.21-py2.py3-none-any.whl (974 kB)
ERROR: Could not find a version that satisfies the requirement vtk (from versions: none)
ERROR: No matching distribution found for vtk
Error: Process completed with exit code 1.
```

*after removing vtk from the YAML file*

Packages will randomly be installed for python 3.9 instead of 3.8, which leads to broken packages.

From https://github.com/espressomd/espresso/pull/3949/checks?check_run_id=1254415500
```
Collecting numpy
  Downloading numpy-1.19.2.zip (7.3 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Collecting cython
  Using cached Cython-0.29.21-py2.py3-none-any.whl (974 kB)
Collecting h5py
  Downloading h5py-2.10.0.tar.gz (301 kB)
Collecting scipy
  Downloading scipy-1.5.2.tar.gz (25.4 MB)
  Installing build dependencies: started
  Installing build dependencies: still running...
  Installing build dependencies: still running...
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Collecting six
  Downloading six-1.15.0-py2.py3-none-any.whl (10 kB)
Building wheels for collected packages: numpy, h5py, scipy
  Building wheel for numpy (PEP 517): started
  Building wheel for numpy (PEP 517): still running...
  Building wheel for numpy (PEP 517): still running...
  Building wheel for numpy (PEP 517): finished with status 'done'
  Created wheel for numpy: filename=numpy-1.19.2-cp39-cp39-macosx_10_15_x86_64.whl size=4625168 sha256=77ce8593974ee9993ce15d4b394c9d1b7820445906eb49919b7ef1a8be502ac4
  Stored in directory: /Users/runner/Library/Caches/pip/wheels/a3/17/dd/f2dba23a35bb6008732772ccfb13d3d0e537fbc6919ce6862b
  Building wheel for h5py (setup.py): started
  Building wheel for h5py (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/local/opt/python@3.9/bin/python3.9 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-futfj5e1/h5py/setup.py'"'"'; __file__='"'"'/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-futfj5e1/h5py/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-wheel-w_tuqnvw
       cwd: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-futfj5e1/h5py/
```

Re-running CI can solve it: https://github.com/espressomd/espresso/runs/1254628766

**Repro steps**  

For the vtk issue:
1. install packages with `pip3 install numpy cython vtk h5py scipy`

For the python 3.9 issue:
1. install packages with `pip3 install numpy cython h5py scipy`
2. re-run the CI pipeline until it installs the packages for python 3.9

Here is the relevant part of our YAML file: https://github.com/espressomd/espresso/blob/cf157f52280adbecbd135ab7476e5b5922664679/.github/workflows/build.yml#L12-L15

**Comment by maxim-lobanov** [comment]
Hello @jngrad , thank you for report and details.
We are working to roll back the broken image.

**Comment by maxim-lobanov** [comment]
Just to provide additional context. We have found that this issue is caused by very strange behavior of `python` brew formula.
If you run the following commands:
1) `brew install python@3.8`
2) `brew install python@3.9`
3) `python3 --version` and `pip3 --version`

After these commands, `python3` and `pip3` will be linked by different formulas: 
```
ac-1972:~ runner$ python3 --version
Python 3.8.6
Mac-1972:~ runner$ pip3 --version
pip 20.2.3 from /usr/local/lib/python3.9/site-packages/pip (python 3.9)
Mac-1972:~ runner$ ls -la /usr/local/bin/python3
lrwxr-xr-x  1 runner  admin  38 Oct 15 01:01 /usr/local/bin/python3 -> ../Cellar/python@3.8/3.8.6/bin/python3
Mac-1972:~ runner$ ls -la /usr/local/bin/pip3
lrwxr-xr-x  1 runner  admin  35 Oct 15 01:02 /usr/local/bin/pip3 -> ../Cellar/python@3.9/3.9.0/bin/pip3
```

We have prepared workaround for this issue in #1824. 
The previous image was rolled back, the new image will be deployed early next week.

**Comment by jngrad** [comment]
Thanks a lot! CI has resumed on our project.

**Comment by al-cheb** [comment]
Hello, @jngrad 
The new image with fixed pip3 point (20201017.1) has been deployed - https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md

```
Pip 19.3.1 (python 2.7)
Pip 20.2.3 (python 3.8)

```
